R-packages/evalcast/R/evaluate_predictions.R

```diff
@@ -1,103 +1,51 @@
+#' error measures.
 #'
-#' @param predictions_cards tibble of predictions
-#'   that are all for the same prediction task, meaning they are for the same
-#'   response, incidence period,and geo type. Forecasts may be for a
-#'   different forecast date or forecaster.
-#'   A predictions card may be created by the function
-#'   [get_predictions()], downloaded with [get_covidhub_predictions()] or
-#'   possibly created manually.
+#' @param predictions_cards tibble of quantile forecasts, which contains at
+#'   least `quantile` and `value` columns, as well as any other prediction task
+#'   identifiers. For covid data, a predictions card may be created by the
+#'   function [get_predictions()], downloaded with [get_covidhub_predictions()]
+#'   or created manually.
 #' @param err_measures Named list of one or more functions, where each function
 #'   takes a data frame with three columns `quantile`, `value` and `actual`
-#'   (i.e., observed) returns a scalar measure of error. Null or an empty list
-#'   may be provided if scoring is not desired.
-#' @param backfill_buffer How many days until response is deemed trustworthy
-#'   enough to be taken as correct? See details for more.
-#' @param side_truth you can optionally provide your own truth data (observed).
-#'   Adding this column bypasses all checks to get appropriate data as
-#'   reported at the time of the forecast. This column should either be a vector
-#'   the same length as `predictions_cards` or a data frame that will be
-#'   joined to `predictions_cards` by all available columns. If a data frame,
-#'   the observed data should be named `actual`
-#' @param grp_vars character vector of named columns in the predictions_cards
-#'  such that the combination gives a unique (quantile) prediction. Ignored if
-#'  `predictions_cards` is of the type returned by [get_covidhub_predictions()]
-#'  or [get_predictions()] (class is `predictions_cards`)
-#' @template geo_type-template 
+#'   (i.e., observed) and returns a scalar measure of error. Null or an empty
+#'   list may be provided if scoring is not desired.
+#' @param truth truth data (observed). This should be a data frame that will be
+#'   joined to `predictions_cards` by all available columns. The observed data
+#'   column should be named `actual`.
+#' @param grp_vars character vector of named columns in `predictions_cards`
+#'  such that the combination gives a unique (quantile) prediction.
 #'
 #' @return tibble of "score cards". Contains the same information as the
-#'   `predictions_cards()` with additional columns for each err_measure and
-#'   for the truth (named `actual`). All the columns may be necessary for
-#'   some types of plotting, but it may be useful to shrink this down
+#'   `predictions_cards()` with additional columns for each `err_measure` and
+#'   for the truth (named `actual`).
 #'
 #' @export
 evaluate_predictions <- function(
   predictions_cards,
+  truth,
```

Comment:
Can we call this something like `truth_data` maybe? Something to emphasize that it needs to be a df with some columns.

Maybe also default to the truth being named "actual" with an option to pass in other names?

Reply:
Updated the object to `truth_data`. Didn't touch the "actual" column name code. Seems large and distinct enough to warrant a separate PR.